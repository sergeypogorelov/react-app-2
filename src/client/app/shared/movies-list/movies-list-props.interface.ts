import { Movie } from '../../core/interfaces/movie/movie.interface';

export interface MoviesListProps {
  movies: Movie[];
}
