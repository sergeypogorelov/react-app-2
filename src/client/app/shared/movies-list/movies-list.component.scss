@import '../../../styles/variables';
@import '../../../styles/functions';

.movies-list {
  &-empty {
    margin-top: 150px;
    font-size: 52px;
    color: whiteColor(0.8);
    text-align: center;
  }
  .movie-wrapper {
    margin-bottom: 50px;
  }
  min-height: 550px;
  padding: 80px 60px 0 60px;
  background-color: $c-background;
}
