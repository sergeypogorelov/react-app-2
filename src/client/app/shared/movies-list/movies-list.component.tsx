import './movies-list.component.scss';

import React, { FunctionComponent } from 'react';

import { MoviesListProps } from './movies-list-props.interface';

import { MovieItem } from './movie-item/movie-item.component';

export const MoviesList: FunctionComponent<MoviesListProps> = (props) => {
  return (
    <div className="movies-list">
      {props.movies.length > 0 ? (
        <div className="row">
          {props.movies.map((i) => (
            <div className="col-4 movie-wrapper" key={i.id}>
              <MovieItem item={i} />
            </div>
          ))}
        </div>
      ) : (
        <h3 className="movies-list-empty">No movies found</h3>
      )}
    </div>
  );
};
