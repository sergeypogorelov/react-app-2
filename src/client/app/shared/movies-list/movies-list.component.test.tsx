import React from 'react';
import renderer from 'react-test-renderer';

import { MoviesList } from './movies-list.component';

test('MoviesList component should render correctly', () => {
  const tree = renderer.create(<MoviesList />).toJSON();
  expect(tree).toMatchSnapshot();
});
