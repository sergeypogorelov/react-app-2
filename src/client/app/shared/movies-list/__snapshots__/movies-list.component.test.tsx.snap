// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component should render correctly 1`] = `
<div
  className="movies-list"
>
  <h3
    className="movies-list-empty"
  >
    No movies found
  </h3>
</div>
`;
